cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResources.java

```diff
@@ -150,18 +163,33 @@ public void deleteDiskEncryptionSet(DiskEncryptionSetDeletionRequest diskEncrypt
         }
     }
 
+    private void checkAndDeleteDesResourceGroupName(AzureClient azureClient, String desResourceGroupName) {
+        Matcher matcher = Pattern.compile(".*-CDP_DES-.*").matcher(desResourceGroupName);
+        if (matcher.matches()) {
+            LOGGER.info("Deleting resource group \"{}\".", desResourceGroupName);
+            azureClient.deleteResourceGroup(desResourceGroupName);
+        } else {
+            LOGGER.info("Deletion of resource group \"{}\" is not required.", desResourceGroupName);
+        }
+    }
+
     private CreatedDiskEncryptionSet getOrCreateDiskEncryptionSetOnCloud(AuthenticatedContext authenticatedContext, AzureClient azureClient,
             String desResourceGroupName, String sourceVaultId, DiskEncryptionSetCreationRequest diskEncryptionSetCreationRequest) {
         CloudContext cloudContext = diskEncryptionSetCreationRequest.getCloudContext();
+        String region = cloudContext.getLocation().getRegion().getRegionName();
+        Map<String, String> tags = diskEncryptionSetCreationRequest.getTags();
         String diskEncryptionSetName = azureUtils.generateDesNameByNameAndId(
                 String.format("%s-DES-", cloudContext.getName()), diskEncryptionSetCreationRequest.getId());
         LOGGER.info("Checking if Disk Encryption Set \"{}\" exists", diskEncryptionSetName);
         DiskEncryptionSetInner createdSet = azureClient.getDiskEncryptionSetByName(desResourceGroupName, diskEncryptionSetName);
         if (createdSet == null) {
+            if (!diskEncryptionSetCreationRequest.isSingleResourceGroup()) {
+                LOGGER.info("Creating resource group \"{}\" for disk encryption set.", desResourceGroupName);
+                azureClient.createResourceGroup(desResourceGroupName, region, tags);
```

Comment:
This call should be fault-resilient. It should first check for the existence of the RG and only create it if absent.